src/SFML/Window/iOS/SFAppDelegate.mm

```diff
@@ -151,6 +162,36 @@ - (void)applicationWillTerminate:(UIApplication *)application
     }
 }
 
+- (bool)supportsOrientation:(UIDeviceOrientation)orientation
+{
+    if (!self.sfWindow)
+        return false;
+    NSArray *supportedOrientations = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"UISupportedInterfaceOrientations"];
+    if (!supportedOrientations)
+        return false;
+    int appFlags = 0;
+    if ([supportedOrientations containsObject:@"UIInterfaceOrientationPortrait"])
+        appFlags += UIInterfaceOrientationMaskPortrait;
+    if ([supportedOrientations containsObject:@"UIInterfaceOrientationPortraitUpsideDown"])
+        appFlags += UIInterfaceOrientationMaskPortraitUpsideDown;
+    if ([supportedOrientations containsObject:@"UIInterfaceOrientationLandscapeLeft"])
+        appFlags += UIInterfaceOrientationMaskLandscapeLeft;
+    if ([supportedOrientations containsObject:@"UIInterfaceOrientationLandscapeRight"])
+        appFlags += UIInterfaceOrientationMaskLandscapeRight;
+    UIViewController* rootViewController = [((__bridge UIWindow*)(self.sfWindow->getSystemHandle())) rootViewController];
+    if (!rootViewController || ![rootViewController shouldAutorotate])
+        return false;
+    return (1 << orientation)
+        & [rootViewController supportedInterfaceOrientations]
+        & appFlags;
```

Comment:
It isn't a boolean expression, it is a bit-operation. && would give wrong results. I fact, using &&, the expression would always be true.